dracut/driver_updates.py

```diff
@@ -206,21 +206,41 @@ def ensure_dir(d):
     """make sure the given directory exists."""
     subprocess.check_call(["mkdir", "-p", d])
 
-def move_files(files, destdir):
+def move_files(files, destdir, basedir):
     """move files into destdir (iff they're not already under destdir)"""
-    ensure_dir(destdir)
     for f in files:
         if f.startswith(destdir):
             continue
-        subprocess.call(["mv", "-f", f, destdir])
+        dest = destdir+"/"+dest_strip(f, basedir)
+        ensure_dir(os.path.dirname(dest))
+        subprocess.call(["mv", "-f", f, dest])
 
-def copy_files(files, destdir):
+def dest_strip(dest, basedir):
+    """strip a base directory plus kernel version from a path"""
+    # Strip the basedir and any leftover leading /'s
+    dest = dest[len(basedir):]
+    while dest.startswith('/'):
+        dest = dest[1:]
+
+    # Look for a leading directory that is a version number
+    if "/" in dest and fnmatch.fnmatch(dest, "*.ko*") and dest[0].isdigit():
+        # Drop the leading directory
+        dest = "/".join(dest.split('/')[1:])
+
+        if dest.startswith("kernel/"):
+            dest = "/".join(dest.split('/')[1:])
+
+    return dest
+
+def copy_files(files, destdir, basedir):
     """copy files into destdir (iff they're not already under destdir)"""
-    ensure_dir(destdir)
     for f in files:
         if f.startswith(destdir):
             continue
-        subprocess.call(["cp", "-a", f, destdir])
+
+        dest = destdir+"/"+dest_strip(f, basedir)
```

Comment:
Same here, but I see this file is a bit special regarding the spaces around operators.